File: JavaScript/ReshapeTheMatrix/ReshapeTheMatrix.js
/**
 * Created by Phoenix-Z on 2017/5/1.
 */
/**
 * (186ms)
 * @param {number[][]} nums
 * @param {number} r
 * @param {number} c
 * @return {number[][]}
 */
var matrixReshape = function(nums, r, c) {
    var m = nums.length, n = nums[0].length;
    if(r *c !== m * n)
        return nums;
    var reshaped = [];
    var now = [];
    for(var i = 0; i < m; i++) {
        for(var j = 0; j < n; j++) {
            now.push(nums[i][j]);
            if(now.length == c) {
                reshaped.push(now);
                now = [];
            }
        }
    }
    return reshaped;
};

File: Java/ReshapeTheMatrix/ReshapeTheMatrix.java
/**
 * (9ms)
 * Created by Phoenix-Z on 2017/5/1.
 */
public class ReshapeTheMatrix {

    public int[][] matrixReshape(int[][] nums, int r, int c) {
        if(nums.length * nums[0].length != r *c)
            return nums;
        int[][] reshaped = new int[r][c];
        int i = 0, j = 0;
        for(int k = 0; k < r; k++) {
            for(int l = 0; l < c; l++) {
                reshaped[k][l] = nums[i][j];
                if(++j == nums[0].length) {
                    i++;
                    j = 0;
                }
            }
        }
        return reshaped;
    }

}
